### PR TITLE
RULEAPI-766 Add documentation and integrity checks for new education rule descriptions format

### PR DESCRIPTION
**Pull request by christophe-zurn-sonarsource**

### CODE REVIEW

File: rspec-tools/rspec_tools/validation/description.py [hunk 1]
```diff
@@ -14,13 +16,41 @@
...
       raise RuleValidationError(f'Rule {rule_language.id} has unconventional header "{name}"')
 
+def validate_how_to_fix_it_subsections(rule_language: LanguageSpecificRule):
+  descr = rule_language.description
+  frameworks_counter = 0
+
+  for h3 in descr.find_all('h3'):
+    name = h3.text.strip()
+    # It is important that the Regex here matches the one used by the analyzers when loading the rules content
+    result = re.search('How to fix it in (?:(?:a|an|the)\\s)?(.*)', name)
```
Comment (roberto-orlandi-sonarsource) [inline rspec-tools/rspec_tools/validation/description.py:38]: Here it should be irrelevant, but just to be consistent with what will be in the analyzer you can modify like this:
```suggestion
    result = re.search('How to fix it in (?:(?:an|a|the)\\s)?(.*)', name)
```

**Review by roberto-orlandi-sonarsource — APPROVED** [review]
LGTM.
There is a huge comment about the documentation where I suggest some small modifications which were suggested by Grammarly :)

File: docs/description.adoc [hunk 2]
```diff
@@ -12,7 +12,116 @@ The following are the only allowed second level titles for H2:
...
+....
+
+Note that each framework-specific _How to fix it?_ subsections must start with a H2 subtitle following the given format:
+`How to fix it in (?:(?:a|an|the)\\s)?(.*)`.
+This is important, as this format will be expected by the analyzers when loading the rule content to recognize the different subsections.
+Furthermore, the display name of the framework has to match an allowed framework
+display name, as defined in <<allowed_framework_names.adoc#,this allowed framework names file>>.
+
+Finally, additional information regarding each of the sections and subsections present in the educational rule description
+can be found https://docs.google.com/document/d/1caqBXTO1r7vftWrNXt_aNVe4rHcB5C6u6oKMBFjbzwo[in this document].
```
[the remark this answers is not in the document]
Comment (christophe-zurn-sonarsource) [reply docs/description.adoc:124]: Ah thanks. Good point, I wanted to do that, and then I forgot :D Thanks :pray:

**Comment by christophe-zurn-sonarsource** [comment]
Updated DTL display name following offline discussion

**Pull request merged**